core/adapters/base.ts

```diff
@@ -22,18 +23,50 @@ export abstract class Adapter {
 
   protected insertInto(target: dataform.ITarget, columns: string[], query: string) {
     return `
-      insert into ${this.resolveTarget(target)}
-      (${columns.join(",")})
-      select ${columns.join(",")}
-      from (${query}) as insertions`;
+insert into ${this.resolveTarget(target)}
+(${columns.join(",")})
+select ${columns.join(",")}
+from (${query}) as insertions`;
   }
 
   protected oppositeTableType(type: string) {
     return this.baseTableType(type) === "table" ? "view" : "table";
   }
 
   protected where(query: string, where: string) {
-    return `select * from (${query}) as subquery
-        where ${where || "true"}`;
+    return `
+  select * from (${query}) as subquery
+    where ${where || "true"}`;
+  }
+
+  protected refreshIncrementCondition(
+    runConfig: dataform.IRunConfig,
+    tableMetadata: dataform.ITableMetadata
+  ) {
+    return runConfig.fullRefresh || !tableMetadata || tableMetadata.type === "view";
+  }
+
+  protected addPreOps(
+    table: dataform.ITable,
+    dataformCoreVersion: string,
+    tasks: Tasks,
+    refreshIncrementCondition: boolean
```

Comment:
I don't think this should be passed in; instead, `addPreOps` should accept `tableMetadata?: dataform.ITableMetadata` and call `refreshIncrementCondition` itself.

(same with `addPostOps`)

Reply:
I had it separate because I was expecting adapters to have different `refreshIncrementCondition` in the future, but no need to separate it preemptively I guess